packages/flutter/lib/src/material/banner.dart

```diff
@@ -201,5 +329,52 @@ class MaterialBanner extends StatelessWidget {
         ],
       ),
     );
+
+    // This provides a static banner during the opt-in phase of the transition to MaterialBanner's new API
+    if (widget.animation == null)
+      return materialBanner;
```

Comment:
Is this early return frowned upon? @Piinks

Reply:
Couldn't find anything on it in the style guide; resolving this comment with the assumption  that it isnt a big deal.